atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -156,6 +173,21 @@ public static SerializableTransactionManager create(
         return create(config, schemas, env, lockServerOptions, allowHiddenTableAccess, UserAgents.DEFAULT_USER_AGENT);
     }
 
+    /**
+     * Create a {@link SerializableTransactionManager} with provided configuration, a set of
+     * {@link Schema}s, {@link LockServerOptions}, and an environment in which to register HTTP server endpoints.
+     */
+    public static SerializableTransactionManager create(
```

Comment:
Agreed that the chain of create() methods is awful. If there's any way of improving its readability, please let me know. :(

This change does implement live reloading if the callers use the new public method
```
    create(
            AtlasDbConfig config,
            java.util.function.Supplier<AtlasDbRuntimeConfig> runtimeConfig,
            Set<Schema> schemas,
            Environment env,
            boolean allowHiddenTableAccess)
```

Since we do not want to make a breaking change, for the callers of the old method we preserve the original behavior and create a constant supplier.